site/content/how-to/monitoring/tracing.md

```diff
@@ -0,0 +1,333 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: coffee
+spec:
+  ports:
+  - port: 80
+    targetPort: 8080
+    protocol: TCP
+    name: http
+  selector:
+    app: coffee
+EOF
+```
+
+Next we'll create the Gateway resource and HTTPRoute for our app:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: cafe
+spec:
+  gatewayClassName: nginx
+  listeners:
+  - name: http
+    port: 80
+    protocol: HTTP
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: coffee
+spec:
+  parentRefs:
+  - name: cafe
+  hostnames:
+  - "cafe.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /coffee
+    backendRefs:
+    - name: coffee
+      port: 80
+EOF
+```
+
+Let's ensure that traffic can flow to our application.
+
+{{< note >}}If you have a DNS record allocated for `cafe.example.com`, you can send the request directly to that hostname, without needing to resolve.{{< /note >}}
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee
+```
+
+We should see a response from the coffee Pod.
+
+```text
+Server address: 10.244.0.69:8080
+Server name: coffee-6b8b6d6486-k5w5w
+URI: /coffee
+```
+
+Assuming that you have access to the [Jaeger dashboard](http://127.0.0.1:16686) from earlier in the guide, you shouldn't see any tracing information yet. This means we need to create our `ObservabilityPolicy`.
+
+### Create the ObservabilityPolicy
+
+To enable tracing for our coffee HTTPRoute, we create the following policy:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.nginx.org/v1alpha1
+kind: ObservabilityPolicy
+metadata:
+  name: coffee
+spec:
+  targetRefs:
+  - group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: coffee
+  tracing:
+    strategy: ratio
+    ratio: 50
+    spanAttributes:
+    - key: coffee-key
+      value: coffee-value
+EOF
+```
+
+This policy attaches to the coffee HTTPRoute and enables ratio-based tracing, where 50% of requests will be sampled. We've also included a span attribute to add extra data to the spans.
+
+Let's check the status of the policy:
+
+```shell
+kubectl describe observabilitypolicies.gateway.nginx.org coffee
+```
+
+```text
+Status:
+  Ancestors:
+    Ancestor Ref:
+      Group:      gateway.networking.k8s.io
+      Kind:       HTTPRoute
+      Name:       coffee
+      Namespace:  default
+    Conditions:
+      Last Transition Time:  2024-05-23T18:13:03Z
+      Message:               Policy is accepted
+      Observed Generation:   1
+      Reason:                Accepted
+      Status:                True
+      Type:                  Accepted
+    Controller Name:         gateway.nginx.org/nginx-gateway-controller
+```
+
+The policy is accepted, so now let's send some more traffic.
+
+```shell
+for i in $(seq 1 10); do curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee; sleep 1; done
```

Comment:
will it work on Windows?

Reply:
Probably not. Think it's worth having a separate command for windows, or just have the curl command and say "run this command 10 times"?

Reply:
> t have the curl command and say "run this command 10 times"?

sounds good
is 10 times necessary? of a few times is enough

Reply:
Nah, I can reduce that